src/composer/ComposerOracle.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.0;
+
+import {IOracle} from "../interfaces/IOracle.sol";
+import {BoringERC20} from "../libraries/BoringERC20.sol";
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+
+import { console2 } from "forge-std/console2.sol";
+
+/**
+ * @title ComposerOracle
+ */
+contract ComposerOracle is IOracle {
+
+    event OracleSet(address indexed base, address indexed quote, IOracle indexed oracle);
+    event PathSet(address indexed base, address indexed quote, address[] indexed path);
+
+    struct OracleWithDecimals {
+        IOracle oracle;
+        uint8 baseDecimals;
+        uint8 quoteDecimals;
```

Comment:
In that case I would have to fit each scalar in an uint48, so that along with the IOracle the struct fits in 32 bytes. That would allow us scalars up to 1e19, which would be a problem now that many erc4626 vaults have a large number of decimals to avoid inflation attacks.